src/pulse3D/excel_writer.py

```diff
@@ -261,16 +296,44 @@ def write_xlsx(plate_recording, name=None):
         well_index = well_file[WELL_INDEX_UUID]
         well_name = TWENTY_FOUR_WELL_PLATE.get_well_name_from_well_index(well_index)
 
+        # find bounding indices with respect to well recording
+        well_start_idx, well_end_idx = truncate(source_series=time_points,
+                                                lower_bound=well_file.force[0][0],
+                                                upper_bound=well_file.force[0][-1])
+
+        # find bounding indices of specified start/end windows
+        window_start_idx, window_end_idx = truncate(source_series=time_points/MICRO_TO_BASE_CONVERSION,
+                                                    lower_bound=start_time,
+                                                    upper_bound=end_time)
+        
+        start_idx = np.max([window_start_idx, well_start_idx])
+        end_idx = np.min([window_end_idx, well_end_idx])
+
+        # fit interpolation function on recorded data
+        interp_data_fn = interpolate.interp1d(well_file.force[0,:], 
+                                              well_file.force[1,:])
+
+        # interpolate, normalize, and scale data
+        interpolated_force = interp_data_fn(time_points[start_idx:end_idx])
+        interpolated_well_data = np.row_stack([time_points[start_idx:end_idx],
+                                               interpolated_force])
+
+        min_value = min(interpolated_force)
+        interpolated_force -= min_value
+        interpolated_force *= MICRO_TO_BASE_CONVERSION
+
         try:
+            # compute peaks / valleys on interpolated well data
             log.info(f"Finding peaks and valleys for well {well_name}")
-            # peaks_and_valleys = peak_detector(well_file.noise_filtered_magnetic_data)
-            peaks_and_valleys = peak_detector(well_file.force)
+            peaks_and_valleys = peak_detector(interpolated_well_data, start_time=start_time, end_time=end_time)
```

Comment:
Should have made a note of this in the PR -- peaks/valleys and metrics are now computed using the interpolated recording data